beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -475,18 +475,18 @@ impl<T: BeaconChainTypes> Critical<T> {
         Ok(())
     }
 
-    /// Removes and returns the pending_components corresponding to
-    /// the `block_root` or `None` if it does not exist
-    pub fn pop_pending_components(
+    /// Returns the pending_components corresponding to  the `block_root` or `None` if it does not
+    /// exist
+    pub fn get_pending_components(
         &mut self,
         block_root: Hash256,
         store: &OverflowStore<T>,
     ) -> Result<Option<PendingComponents<T::EthSpec>>, AvailabilityCheckError> {
-        match self.in_memory.pop_entry(&block_root) {
-            Some((_, pending_components)) => Ok(Some(pending_components)),
+        match self.in_memory.get(&block_root) {
```

Comment:
I think there might be an rare edge case scenario here where:

1. we load the `PendingComponents` from memory
2. we update it and try to put it back, however `in_memory` is at capacity and it goes to the disk.
3. next time we want to update it, it loads from memory which would give us the outdated component

In this case I think it would just delay availability as we would do another lookup right? I think we might be able to avoid this altogether if we check if the components is already in memory before we attempt to use the overflow store?

Another one that's probably less of an issue:

1. We load the `PendingComponents`, not found in memory, so it returned from `store`
2. We update it and put it back to memory, now we have two versions one in memory and one on disk
3. Now if the in memory one somehow gets pruned first, then we end up loading an old component from disk.

I think this is unlikely under normal network condition as the LRU cache's default capacity is `1024`.